lib/streamlit/elements/arrow_altair.py

```diff
@@ -759,9 +769,9 @@ def altair_chart(
         use_container_width: bool = False,
         theme: Literal["streamlit"] | None = "streamlit",
         # TODO(willhuang1997): This will need to be finalized to "rerun" / "ignore" or True / False
-        on_select: Union[str, Callable[..., None], True, False, None] = None,
+        on_select: Union[str, Callable[..., None], bool, None] = None,
         key: str | None = None,
-    ) -> Union["DeltaGenerator", Dict]:
+    ) -> Union["DeltaGenerator", Dict[Any, Any]]:
```

Comment:
Hmm, we probably want to overload the function signature to only return the `Dict[Any, Any]` if its actually activated with `on_select`. And in all other cases have `DeltaGenerator` as return value.